playbooks/common/openshift-node/config.yml

```diff
@@ -84,6 +87,7 @@
     args:
       creates: "{{ etcd_generated_certs_dir }}/{{ item.etcd_cert_subdir }}.tgz"
     with_items: etcd_needing_client_certs
+    when: etcd_client_flannel_certs_missing is defined and etcd_client_flannel_certs_missing
```

Comment:
Yeah, I tested this with external etcd no flannel, external etcd with flannel and then embedded and it works afaict. I'll throw that together to get us working - definitely needs a refactor.